{{cookiecutter.app_name}}/bin/wait-for-postgres.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+set -e
+
+until psql $DATABASE_URL -c "\q" > /dev/null 2>&1; do
+  >&2 echo "Postgres is unavailable - sleeping"
+  sleep 1
+done
```

Comment:
Note that in Docker Compose when booting up the DB service, the following options can be added to ensure the readiness of the DB:

```
db:
  image: postgres:11-alpine
  ports:
    - 5432:5432
  env:
    ...
  options: >-
    --health-cmd pg_isready
    --health-interval 10s
    --health-timeout 5s
    --health-retries 5
```

Also, we should probably migrate the Github Action test workflow to use Docker Compose (there is already the test compose file) instead of running the tests outside of a Docker container. It would allow using the options in both test and dev environments.

Reply:
As discussed on compass, I prefer to use `wait-for-migrate` script over docker for the following reasons.

Run the test in docker compare with host machine on CI.

Advantage
- We can use options in docker-compose

Disadvantage
- Code duplication (If we prefer to run on host machine for local development, `wait-for-postgres` script still needed.)
- Slower because we need build the image step.

What do you think?

Reply:
> If we prefer to run on host machine for local development

The template already has Docker Compose and `env-setup` actually uses it so the choice of running the DB in the host machine is not really recommended 🤔 

In terms of speed, it's slower on the first run but subsequent builds for the same branch is speed up using cache. Read more in this article (teh author is actually Thai 🇹🇭 ): https://dev.to/dtinth/caching-docker-builds-in-github-actions-which-approach-is-the-fastest-a-research-18ei

The current implementation work so it's not a blocker. So reviewing if running tests in Docker on Github Actions can be done in another task and for another version.

Reply:
With the current implementation, DB is running in Docker and application is running on host machine.

Running the test in Docker on CI is created as a new task https://github.com/nimblehq/gin-templates/issues/34